docker/src/main/java/brooklyn/entity/container/docker/registry/DockerRegistryImpl.java

```diff
@@ -0,0 +1,110 @@
+package brooklyn.entity.container.docker.registry;
+
+
+import static org.apache.brooklyn.util.ssh.BashCommands.chain;
+
+import java.net.URI;
+import java.util.Arrays;
+import java.util.List;
+
+import com.google.common.base.Functions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.net.HostAndPort;
+
+import org.apache.brooklyn.core.entity.Attributes;
+import org.apache.brooklyn.core.location.access.BrooklynAccessUtils;
+import org.apache.brooklyn.entity.software.base.SoftwareProcess;
+import org.apache.brooklyn.feed.http.HttpFeed;
+import org.apache.brooklyn.feed.http.HttpPollConfig;
+import org.apache.brooklyn.feed.http.HttpValueFunctions;
+import org.apache.brooklyn.feed.http.JsonFunctions;
+import org.apache.brooklyn.location.ssh.SshMachineLocation;
+import org.apache.brooklyn.util.collections.MutableMap;
+import org.apache.brooklyn.util.guava.Functionals;
+import org.apache.brooklyn.util.os.Os;
+import org.apache.brooklyn.util.ssh.BashCommands;
+import org.apache.brooklyn.util.time.Duration;
+
+import brooklyn.entity.container.DockerAttributes;
+import brooklyn.entity.container.docker.DockerHost;
+import brooklyn.entity.container.docker.DockerInfrastructure;
+import brooklyn.entity.container.docker.application.VanillaDockerApplicationImpl;
+
+public class DockerRegistryImpl extends VanillaDockerApplicationImpl implements DockerRegistry {
+
+    private volatile HttpFeed httpFeed;
+    private final String CREATE_CERTS_SCRIPT_NAME = "docker-registry-create-certs.sh";
+
+    private String getSSHMachineInstallDir() {
+        return config().get(DockerRegistry.DOCKER_HOST).sensors().get(SoftwareProcess.INSTALL_DIR);
+    }
+
+    @Override
+    public void connectSensors() {
+        super.connectSensors();
+        connectServiceUpIsRunning();
+
+        HostAndPort hostAndPort = BrooklynAccessUtils.getBrooklynAccessibleAddress(this, config().get(DockerRegistry.DOCKER_REGISTRY_PORT));
+        sensors().set(Attributes.MAIN_URI, URI.create("https://" + hostAndPort + "/v2"));
+
+
+        httpFeed = HttpFeed.builder()
+                .entity(this)
+                .period(Duration.seconds(3))
+                .baseUri(getAttribute(Attributes.MAIN_URI))
+                .poll(new HttpPollConfig<Boolean>(Attributes.SERVICE_UP)
+                        .onSuccess(Functions.constant(true))
+                        .onFailureOrException(Functions.constant(false)))
+                .poll(new HttpPollConfig<List<String>>(DOCKER_REGISTRY_CATALOG)
+                        .suburl("/_catalog")
+                        .onSuccess(Functionals.chain(
+                                HttpValueFunctions.jsonContents(),
+                                JsonFunctions.walk("repositories"),
+                                JsonFunctions.forEach(JsonFunctions.cast(String.class))))
+                        .onFailureOrException(Functions.constant(Arrays.asList(new String[0]))))
```

Comment:
Maybe `Lists.<String>newArrayList()` instead of creating the empty `String[0]` array?